File: src/modules/update.rs
use crate::args::UpdateArgs;
use crate::utils::status_msg::error;
use std::process::Command;

pub fn update(args: &UpdateArgs) {
    if args.system {
        update_system();
    }

    if args.user {
        update_user();
    }
}

fn update_user() {
    let mut update_user_cmd = Command::new("nebula")
        .arg("update-system")
        .spawn()
        .expect("Failed to start updating the user");

    let status = update_user_cmd.wait().expect("Failed to update the user");

    if !status.success() {
        error("Failed to update the user!");
    }
}

fn update_system() {
    let mut update_system_cmd = Command::new("sudo")
        .args(["nebula", "update-system"])
        .spawn()
        .expect("Failed to start updating the system");

    let status = update_system_cmd
        .wait()
        .expect("Failed to update the system");

    if !status.success() {
        error("Failed to update the system!");
    }
}

File: cmd/update.go
package cmd

import (
	"os"
	"os/exec"

	"github.com/darkmatterorg/orbit/utils"
	"github.com/spf13/cobra"
)

var (
	systemFlag bool
	userFlag   bool

	updateCmd = &cobra.Command{
		Use:   "update",
		Short: "Run an update on user, system or both",
		Run: func(cmd *cobra.Command, args []string) {
			if !utils.IsCmdInstalled("nebula") {
				utils.Error("Nebula needs to be installed")
				return
			}

			if systemFlag {
				utils.Info("Updating system")
				updateSystem()
			}

			if userFlag {
				utils.Info("Updating user")
				updateUser()
			}
		},
	}
)

func init() {
	updateCmd.Flags().BoolVarP(&systemFlag, "system", "s", false, "")
	updateCmd.Flags().BoolVarP(&userFlag, "user", "u", false, "")
	updateCmd.MarkFlagsOneRequired("system", "user")

	RootCmd.AddCommand(updateCmd)
}

func updateSystem() {
	cmdToRun := exec.Command("sudo", "nebula", "update-system")

	cmdToRun.Stdout = os.Stdout

	if err := cmdToRun.Run(); err != nil {
		utils.Error("Failed to update the system")
		os.Exit(1)
	}
}

func updateUser() {
	cmdToRun := exec.Command("nebula", "update-system")

	cmdToRun.Stdout = os.Stdout

	if err := cmdToRun.Run(); err != nil {
		utils.Error("Failed to update the user")
		os.Exit(1)
	}
}
